core/src/main/java/org/jboss/set/mjolnir/archive/GitDeleteRepository.java

```diff
@@ -0,0 +1,43 @@
+package org.jboss.set.mjolnir.archive;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.ListBranchCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.internal.storage.file.FileRepository;
+import org.eclipse.jgit.lib.Ref;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+public class GitDeleteRepository {
+
+    private String PATH;
+    private String gitUserName;
+
+    public GitDeleteRepository(String path, String gitUserName) {
+        this.PATH = path;
+        this.gitUserName = gitUserName;
+
+    }
+
+    public void gitRemoveBranches() throws GitAPIException, IOException {
+        File gitDir = new File(PATH);
+        List<Ref> refs = Git.open(gitDir)
+                .branchList().setListMode(ListBranchCommand.ListMode.REMOTE)
+                .call();
```

Comment:
Is it possible to extract this GIT call (reading branches) into a separate method in `GitArchiveRepository` class? So that `GitArchiveRepository` can be a wrapper containing all the functionalities related to manipulation of local git repositories...

Reply:
Added into **GitArchiveRepository**